src/lexer.d

```diff
@@ -447,26 +447,26 @@ public:
                         }
                         if (id == Id.DATE)
                         {
-                            t.ustring = cast(char*)date;
+                            t.ustring = cast(const(char)*)date;
```

Comment:
These only had a case because of autoconversion.  They should use `date.ptr` instead and implicitly convert.